app/src/main/java/ch/epfl/skysync/components/forms/SearchBarCustom.kt

```diff
@@ -0,0 +1,117 @@
+package ch.epfl.skysync.components.forms
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun <T> SearchBarCustom(
+    query: String,
+    onQueryChange: (String) -> Unit,
+    onSearch: (String) -> Unit,
+    active: Boolean,
+    onActiveChange: (Boolean) -> Unit,
+    onElementClick: (T) -> Unit,
+    propositions: List<T>,
+    showProposition: (T) -> String,
+    placeholder: String = "Search"
+) {
+  val focusRequester = remember { FocusRequester() }
+  var height by remember { mutableStateOf(90.dp) }
+  val animatedHeight by animateDpAsState(targetValue = height, label = "")
+  val keyboardController = LocalSoftwareKeyboardController.current
+  LaunchedEffect(active) { height = if (active) 300.dp else 90.dp }
```

Comment:
You could do like the dropdownmenu and make the height modular

Reply:
The height change is necessary to make the difference between the active and inactive states. Also more than 300.dp when it is opened it not really usefull as there is the keyboard that will mask it and we won't be able to click on it.